meta-webserver/recipes-webadmin/cockpit/cockpit_217.bb

```diff
@@ -0,0 +1,178 @@
+
+FILES_${PN}-machines = " \
+    ${datadir}/cockpit/machines \
+    ${datadir}/metainfo/org.cockpit-project.cockpit-machines.metainfo.xml \
+"
+FILES_${PN}-selinux = " \
+    ${datadir}/cockpit/selinux \
+    ${datadir}/metainfo/org.cockpit-project.cockpit-selinux.metainfo.xml \
+"
+FILES_${PN}-playground = "${datadir}/cockpit/playground"
+FILES_${PN}-docker = " \
+    ${datadir}/cockpit/docker \
+    ${datadir}/metainfo/org.cockpit-project.cockpit-docker.metainfo.xml \
+"
+FILES_${PN}-dashboard = "${datadir}/cockpit/dashboard"
+ALLOW_EMPTY_${PN}-dashboard = "1"
+
+FILES_${PN}-bridge = " \
+    ${bindir}/cockpit-bridge \
+    ${libexec}/cockpit-askpass \
+"
+RDEPENDS_${PN}-bridge = ""
+
+FILES_${PN}-desktop = "${libexecdir}/cockpit-desktop"
+RDEPENDS_${PN}-desktop += "bash"
+
+FILES_${PN}-ws = " \
+    ${sysconfdir}/cockpit/ws-certs.d \
+    ${sysconfdir}/pam.d/cockpit \
+    ${sysconfdir}/issue.d/cockpit.issue \
+    ${sysconfdir}/motd.d/cockpit \
+    ${datadir}/cockpit/motd/update-motd \
+    ${datadir}/cockpit/motd/inactive.motd \
+    ${systemd_system_unitdir}/cockpit.service \
+    ${systemd_system_unitdir}/cockpit-motd.service \
+    ${systemd_system_unitdir}/cockpit.socket \
+    ${systemd_system_unitdir}/cockpit-wsinstance-http.socket \
+    ${systemd_system_unitdir}/cockpit-wsinstance-http.service \
+    ${systemd_system_unitdir}/cockpit-wsinstance-http-redirect.socket \
+    ${systemd_system_unitdir}/cockpit-wsinstance-http-redirect.service \
+    ${systemd_system_unitdir}/cockpit-wsinstance-https-factory.socket \
+    ${systemd_system_unitdir}/cockpit-wsinstance-https-factory@.service \
+    ${systemd_system_unitdir}/cockpit-wsinstance-https@.socket \
+    ${systemd_system_unitdir}/cockpit-wsinstance-https@.service \
+    ${systemd_system_unitdir}/system-cockpithttps.slice \
+    ${libdir}/tmpfiles.d/cockpit-tempfiles.conf \
+    ${sbindir}/remotectl \
+    ${libdir}/security/pam_ssh_add.so \
+    ${libdir}/security/pam_cockpit_cert.so \
+    ${libexecdir}/cockpit-ws \
+    ${libexecdir}/cockpit-wsinstance-factory \
+    ${libexecdir}/cockpit-tls \
+    ${libexecdir}/cockpit-session \
+    ${localstatedir}/lib/cockpit \
+    ${datadir}/cockpit/static \
+    ${datadir}/cockpit/branding \
+"
+CONFFILES_${PN}-ws += " \
+    ${sysconfdir}/issue.d/cockpit.issue \
+    ${sysconfdir}/motd.d/cockpit \
+"
+RDEPENDS_${PN}-ws += "openssl-bin"
+SYSTEMD_SERVICE_${PN}-ws = "cockpit.socket"
+
+FILES_${PN} += " \
+    ${datadir}/cockpit/base1 \
+    ${sysconfdir}/cockpit/machines.d \
+    ${datadir}/polkit-1/actions/org.cockpit-project.cockpit-bridge.policy \
+    ${datadir}/cockpit/ssh \
+    ${libexecdir}/cockpit-ssh \
+    ${datadir}/cockpit \
+    ${datadir}/metainfo/cockpit.appdata.xml \
+    ${datadir}/pixmaps/cockpit.png \
+"
+RDEPENDS_${PN} += "${PN}-bridge"
+
+do_install_append() {
+    pkgdatadir=${datadir}/cockpit
+
+    # avoid host contamination
+    find ${D}${datadir}/cockpit -name manifest.json -exec chown root:root {} \;
```

Comment:
root seems to be iffy perhaps look into install target of the package and see if they are not using cp -a or some such

Reply:
Unfortunately I have not yet found the cause of this.

Reply:
In the install target, the files are copied using `tar` which leads to host contamination.
I have a patch for that I could add to a subsequent PR.

Reply:
That's great!
If desired you can make a pull request on https://github.com/hoinmic/meta-openembedded . Then we could bring everything upstream at once.

Reply:
@eroullit its fine, please prepare a patch on top of master-next which already has this patch. and send a new pull.